transport/http/src/main/java/org/kaazing/gateway/transport/http/HttpConnector.java

```diff
@@ -386,6 +406,19 @@ public DefaultHttpSession call() throws Exception {
                                 parentEx,
                                 new HttpBufferAllocator(parentAllocator));
                         parent.setAttribute(HTTP_SESSION_KEY, httpSession);
+
+                        // Set X-FORWARDED-FOR header on the HTTP session.
```

Comment:
I think the following could be wrong. This would add the IP address of the origin server. Run any of the http proxy tests and check the port to verify this. IIRC, the httpsession is created even before it sent the http request to the origin server (and in this case tcp address would origin server's address).